lib/gosu.rb

```diff
@@ -22,16 +22,17 @@ module Gosu
   extend FFI::Library
   ffi_lib Gosu::LIBRARY_PATH
 
-  callback :_callback,            [],         :void
-  callback :_callback_with_block, [:pointer], :void
+  callback :_callback,             [],         :void
+  callback :_callback_with_block,  [:pointer], :void
+  callback :_callback_with_string, [:string],  :void
```

Comment:
Actually...shouldn't it be `[:pointer, :string]`? And do we even have callbacks without any arguments at all? Why does this work? 🤔

Reply:
Technically, yes... it should be `[:pointer, :string]`. If I leave the argument list empty for all those callbacks, `rake test` still succeeds. 🤔

Reply:
I've changed it to `[:pointer, :string]` just to be on the safe side. Let me know what you think, I'm not sure if the longer name in the TextInput wrapper is worth it. The FFI definitions lead to some pretty long code lines.

Reply:
I was thinking of possibly removing the method rename argument from the ffi function definitions to shorten them a bit
```ruby
attach_function :_image_load_tiles_from_image, :Gosu_Image_create_tiles_from_image,  [:pointer, :int, :int, :_callback_with_image, :pointer, :uint32],  :void
```
to
```ruby
attach_function :Gosu_Image_create_tiles_from_image,  [:pointer, :int, :int, :_callback_with_image, :pointer, :uint32], :void
```

What do your think? This would mean calling the function would be `Gosu_Image_load_tiles_from_image` instead of `_image_load_tiles_from_image`

Reply:
Less repetition sounds great to me :+1: Should I do anything in this PR? (Which is a bit of a mixed bag already)

Reply:
If you can easily do a mass rename I'd say go for it. Otherwise, I can merge this and we chip away at renaming them.

Reply:
Hmm, I think I'd rather merge this PR first, and then combine the mass renames with the FFI error handling adjustments.